src/menu/menu.c

```diff
@@ -253,46 +182,34 @@ item_create(struct menu *menu, const char *text, bool show_arrow)
 	/* Font nodes */
 	menuitem->normal.buffer = scaled_font_buffer_create(menuitem->normal.tree);
 	menuitem->selected.buffer = scaled_font_buffer_create(menuitem->selected.tree);
-	if (!menuitem->normal.buffer || !menuitem->selected.buffer) {
-		wlr_log(WLR_ERROR, "Failed to create menu item '%s'", text);
-		/*
-		 * Destroying the root node will destroy everything,
-		 * including the node descriptor and scaled_font_buffers.
-		 */
-		wlr_scene_node_destroy(&menuitem->tree->node);
-		free(menuitem);
-		return NULL;
-	}
+	assert(menuitem->normal.buffer && menuitem->selected.buffer);
 	menuitem->normal.text = &menuitem->normal.buffer->scene_buffer->node;
 	menuitem->selected.text = &menuitem->selected.buffer->scene_buffer->node;
 
 	/* Font buffers */
-	scaled_font_buffer_update(menuitem->normal.buffer, text, menuitem->native_width,
-		&rc.font_menuitem, theme->menu_items_text_color,
-		theme->menu_items_bg_color, arrow);
-	scaled_font_buffer_update(menuitem->selected.buffer, text, menuitem->native_width,
-		&rc.font_menuitem, theme->menu_items_active_text_color,
-		theme->menu_items_active_bg_color, arrow);
+	scaled_font_buffer_update(menuitem->normal.buffer, menuitem->text,
+		menuitem->native_width, &rc.font_menuitem,
+		theme->menu_items_text_color,
+		theme->menu_items_bg_color, menuitem->arrow);
+	scaled_font_buffer_update(menuitem->selected.buffer, menuitem->text,
+		menuitem->native_width, &rc.font_menuitem,
+		theme->menu_items_active_text_color,
+		theme->menu_items_active_bg_color, menuitem->arrow);
+	scaled_font_buffer_set_max_width(menuitem->normal.buffer,
+		menu->size.width - 2 * theme->menu_items_padding_x);
+	scaled_font_buffer_set_max_width(menuitem->selected.buffer,
+		menu->size.width - 2 * theme->menu_items_padding_x);
```

Comment:
These calls are redundant and force a new render. Also I am slightly confused why the `max_width` argument in `scaled_font_buffer_update()` is different to the one in `scaled_front_buffer_set_max_width()`.

Reply:
Yeah, it's my mistake. I carelessly copied and modified the code, so I didn't realize `scaled_font_buffer_update()` contains `max_width`. I'll open a PR to fix it.